services/frontendhost/frontend.go

```diff
@@ -303,13 +303,38 @@ func convertCGZoneConfigToInternal(cgZoneCfg *c.ConsumerGroupZoneConfig) *shared
 
 // convertCreateCGRequestToInternal converts Cherami CreateConsumerGroupRequest to internal shared CreateConsumerGroupRequest
 func convertCreateCGRequestToInternal(createRequest *c.CreateConsumerGroupRequest) *shared.CreateConsumerGroupRequest {
+
+	// detect and correct the units for 'startFrom' (expected internally to be in nanoseconds)
+	startFrom := createRequest.GetStartFrom()
+
+	// Note: 2000-01-01T00:00 is 946684800 seconds since epoch
+	switch {
+	case startFrom == 0:
+		// special-case '0' and let it through
+
+	case startFrom < 946684800000:
```

Comment:
> Why not use common.MaxUint32 for seconds boundary? This goes to year 2038.

We have some values that are small, but valid. First is 0, which is a special value to mean 'now'. The second is that our documentation suggests that a very small time (e.g. '1') should be used to start from the oldest available messages. Since our CLI interprets its parameters as seconds, '1' second after epoch becomes '1E9' nanoseconds after epoch. We are currently at ~ 1.49E9 seconds after epoch, so for Kiran's scheme to work in the face of our CLI's deficiencies, we need to choose our cutoff points wisely.

Alternatively, we could also amend our CLI to treat small values (e.g. < 1E9) literally, rather than as seconds after epoch, which would simplify the model here.